### PR TITLE
fix(donate): modal opens at most once per session

### LINKED ISSUE #36619: [.dev] donate/support modal shows up a lot
I believe you need to be signed in
Go to a challenge
Complete and submit challenges until the donate/support our non profit prompt shows up
Close it
Then use the back and fourth buttons in the challenge description area
The prompt will pop up on every move to another challenge

It's a little annoying.

Additionally, the donate prompt doesn't show up at all if you aren't signed in.

### PR DESCRIPTION
**Pull request by lancerutkin**
- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [X] My changes do not use shortened URLs or affiliate links.

Closes #36619

### CODE REVIEW

File: client/src/redux/index.js [hunk 1]
```diff
@@ -260,7 +263,8 @@ export const reducer = handleActions(
     }),
     [types.openDonationModal]: state => ({
       ...state,
-      showDonationModal: true
+      showDonationModal: true,
+      donationRequested: true
```
Comment (ahmadabdolsaheb) [inline client/src/redux/index.js]: it would be optimal to have a separate action that switches the donationRequested: true.

**Comment by ahmadabdolsaheb** [comment]
Thanks for the PR  @lancerutkin. 
@raisedadead, LGTM. just the design decision mentioned in the review.

**Comment by lancerutkin** [comment]
@ahmadabdolsaheb I've got that design change ready - let me know if I should push that to this branch.

**Comment by ahmadabdolsaheb** [comment]
@lancerutkin yes please.

**Review by ahmaxed — APPROVED** [review]

**Comment by ahmadabdolsaheb** [comment]
@raisedadead LGTM

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **AWESOME! LGTM.** :sparkles: :sparkles: 

Thanks @lancerutkin, for this pull request and contributing to the code-base for the first time.
We will be looking forward for more contributions in future.

Cheers and happy contributing!

**Pull request merged**